common/djangoapps/track/views/__init__.py

```diff
@@ -111,27 +106,14 @@ def server_track(request, event_type, event, page=None):
     except:
         username = "anonymous"
 
-    # define output:
-    event = {
-        "username": username,
-        "ip": _get_request_ip(request),
-        "referer": _get_request_header(request, 'HTTP_REFERER'),
-        "accept_language": _get_request_header(request, 'HTTP_ACCEPT_LANGUAGE'),
-        "event_source": "server",
-        "event_type": event_type,
-        "event": event,
-        "agent": _get_request_header(request, 'HTTP_USER_AGENT').encode().decode('latin1'),
-        "page": page,
-        "time": datetime.datetime.utcnow().replace(tzinfo=pytz.utc),
-        "host": _get_request_header(request, 'SERVER_NAME'),
-        "context": eventtracker.get_tracker().resolve_context(),
-    }
-
-    # Some duplicated fields are passed into event-tracking via the context by track.middleware.
-    # Remove them from the event here since they are captured elsewhere.
-    shim.remove_shim_context(event)
-
-    log_event(event)
+    context_override = contexts.course_context_from_url(page)
```

Comment:
Change this to simply:

```python
context_override = {}
```

page is often not a URL for these calls (e.g. "x_module"), so calling course_context_from_url here will just result in an override that wipes out the correct value set by the middleware here: https://github.com/edx/edx-platform/blob/61e1eda20df2825a409db3e2d36c69d7c36d3e2d/common/djangoapps/track/middleware.py#L160

Reply:
I suppose we could also try to parse the page as a URL and then just remove the empty "course_id" and "org_id" keys if the parsing fails for whatever reason:

```python
context_override = contexts.course_context_from_url(page)
if context_override.get('course_id', '') == '':
  context_override = {}
```

That actually might be safer. I'm not sure if this function is ever called with a valid URL for the `page` parameter.